### PR TITLE
sof_sdw: Code clean up

### PR DESCRIPTION
**Pull request by charleskeepax**
Some minor refactoring to attempt to reduce the amount of state being passed around in the machine driver and logically group some things a little better.

### CODE REVIEW

File: sound/soc/intel/boards/sof_sdw.c
Comment (plbossart) [inline sound/soc/intel/boards/sof_sdw.c]: does this really make the code better/easier to read?

codec_idx is a counter, while codec_index is assigned line 1421. It looks like a different concept.

@RanderWang @bardliao please chime in on this one.

File: sound/soc/intel/boards/sof_sdw.c
Comment (plbossart) [inline sound/soc/intel/boards/sof_sdw.c]: I'll let @bardliao  and @RanderWang review this less-than-simple piece of code :-)

File: sound/soc/intel/boards/sof_sdw.c
Comment (plbossart) [inline sound/soc/intel/boards/sof_sdw.c]: usually when there is an initialization we use a different line.

I would move all the initialized ints together.

File: sound/soc/intel/boards/sof_sdw.c
Comment (plbossart) [inline sound/soc/intel/boards/sof_sdw.c]: i think we changed this recently, didn't we? @bardliao added support for RT712 with 2 endpoints.

File: sound/soc/intel/boards/sof_sdw.c
Comment (plbossart) [inline sound/soc/intel/boards/sof_sdw.c]: I personally don't like 'id', it could be the dai id, the be id, something else. Using the be_ prefix was intentional to avoid confusions.

File: sound/soc/intel/boards/sof_sdw.c
Comment (plbossart) [inline sound/soc/intel/boards/sof_sdw.c]: didn't morimoto-san add something to better handle DLCs recently? I find the '2' above confusing with the two lines above.

**Review by plbossart — COMMENTED** [review]
Thank you so much @charleskeepax for starting this much-needed cleanup. I only added minor comments below, I think only @bardliao and @RanderWang understand the code in depth, they should be the main reviewers on this.

File: sound/soc/intel/boards/sof_sdw.c
Comment (charleskeepax) [reply sound/soc/intel/boards/sof_sdw.c]: Hmm... it appears he did this patch might be bogus, I will double check.

File: sound/soc/intel/boards/sof_sdw.c
Comment (charleskeepax) [reply sound/soc/intel/boards/sof_sdw.c]: Happy to keep it that way.

File: sound/soc/intel/boards/sof_sdw.c
Comment (charleskeepax) [reply sound/soc/intel/boards/sof_sdw.c]: One is the index into the codec_info array, the other is the index into the codecs dai_link_comps array. I am happy to keep them separate if we prefer but renaming them so they arn't just called the same thing. (idx=index)

File: sound/soc/intel/boards/sof_sdw.c
Comment (charleskeepax) [reply sound/soc/intel/boards/sof_sdw.c]: I can see that rt712 was added with 2 endpoints, but as far as I can see no part of the machine driver accesses anything other than the first endpoint. My assumption is this is still a work in progress and whilst the endpoints are defined at the moment no support is added for the second one. I am happy to drop this patch if support is incoming shortly?

File: sound/soc/intel/boards/sof_sdw.c
Comment (bardliao) [reply sound/soc/intel/boards/sof_sdw.c]: @plbossart @charleskeepax Both of you are right. However, codec_idx is more like an offset to construct the index into the codecs dai_link_comps array. We may rename it to something like codec_offset. But considering that codec_idx is only used in the loop, and codec_index is only used after line 1421, I am fine with combining them together.

File: sound/soc/intel/boards/sof_sdw.c
Comment (bardliao) [inline sound/soc/intel/boards/sof_sdw.c]: Removing the test doesn't look right to me. We only want to add cpu_dai (adr_link) with the endpoint in the same group.
`cpu_dai_id[index++] = ffs(adr_link->mask) - 1;` in line 1302 will add all cpu_dai after the adr_link.
Take tgl_3_in_1_sdca[] for example,
```
static const struct snd_soc_acpi_link_adr tgl_3_in_1_sdca[] = {
        {
                .mask = BIT(0),
                .num_adr = ARRAY_SIZE(rt711_sdca_0_adr),
                .adr_d = rt711_sdca_0_adr,
        },
        {
                .mask = BIT(1),
                .num_adr = ARRAY_SIZE(rt1316_1_group1_adr),
                .adr_d = rt1316_1_group1_adr,
        },
        {
                .mask = BIT(2),
                .num_adr = ARRAY_SIZE(rt1316_2_group1_adr),
                .adr_d = rt1316_2_group1_adr,
        },
        {
                .mask = BIT(3),
                .num_adr = ARRAY_SIZE(rt714_3_adr),
                .adr_d = rt714_3_adr,
        },
        {}
};
```
the 4th adr_link (rt714_3) will be handled, and cpu_dai_id index will be 3 when `get_slave_info` is called for the aggregated rt1316

File: sound/soc/intel/boards/sof_sdw.c
Comment (bardliao) [inline sound/soc/intel/boards/sof_sdw.c]: We probably don't need 
```

	adr_link = mach_params->links;
	if (!adr_link)
		return -EINVAL;
```
at line 1634 any more.

File: sound/soc/intel/boards/sof_sdw.c
Comment (charleskeepax) [reply sound/soc/intel/boards/sof_sdw.c]: Yeah I think you are correct here, good spot. I will rework.

File: sound/soc/intel/boards/sof_sdw.c
Comment (bardliao) [reply sound/soc/intel/boards/sof_sdw.c]: Indeed, we only check the first endpoint in machine driver. Currently, we only care about if the endpoint is aggregated or not. For those codecs with multiple endpoints, we expect they are all non-aggregated. Maybe show the warning message only when it is with multiple endpoints, and any of the endpoint is aggregated?
Does it make sense? @charleskeepax @plbossart

Comment (plbossart) [reply sound/soc/intel/boards/sof_sdw.c]: the concept of endpoint was a placeholder for future ACPI-based descriptions on the hardware. Indeed if a codec exposes multiple endpoints, the idea is that they can be used independently for different usages. The problem is that ACPI work has been put on hold for what looks like forever and I am not sure there's any merit in trying to improve these endpoints. Better to leave them as is and maybe we'll remove all of this at some point.

File: sound/soc/intel/boards/sof_sdw.c
Comment (RanderWang) [reply sound/soc/intel/boards/sof_sdw.c]: codec_idx has different meaning with codec_index as Bard said. I don't prefer to mix them.

**Review by RanderWang — COMMENTED** [review]
Most part is good for me. Thanks!

**Comment by charleskeepax** [comment]
Sorry about the delay on updating this. I decided perhaps it would be better to split this pull request into a couple of pulls to ease review a little. I have trimmed this one down to just be the fairly trivial (and hopefully not too controversial) clean up patches at the start of the chain. Once we have those out the way I will follow up with some of the more heavy refactoring changes.

**Review by bardliao — APPROVED** [review]
LGTM, thanks

**Review by plbossart — APPROVED** [review]
Thanks @charleskeepax !

**Review by ranj063 — APPROVED** [review]

**Review by RanderWang — APPROVED** [review]

**Pull request merged**
